### ISSUE TITLE
Document some patterns for scroll effects

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#18078]
If you want to space out some widgets, except if you run out of room you want it to scroll:

- Wrap a SingleChildScrollView in a LayoutBuilder, then inside the SingleChildScrollView put a ConstrainedBox whose minHeight is the maxHeight constraint LayoutBuilder saw, and as a child of the ConstrainedBox put your Column with the flexible contents, with its mainAxisSize set to MainAxisSize.min.

To take up all remaining space, use a SliverFillRemaining, or, if it will almost always always be visible:

- A LayoutBuilder, to get the size of the general area, child: The SingleChildScrollView so that the contents can scroll, child: A ConstrainedBox that sets a minimum constraint of the maxHeight from the LayoutBuilder's constraints, child: An IntrinsicHeight widget that forces the constraints to be bounded, in theory tightly to the Column's contents' height, but forced by the ConstrainedBox to be at least as high as the viewport, child: A Column with the children, one or more of which is Expanded.

**Comment by Sfshaza** [comment]
@brianegan, a good candidate for the cookbook?

**Comment by brianegan** [comment]
@Sfshaza Yep, sounds good! I've got a few tutorials on ListView right now, and think it might make sense to add an "Advanced Lists" or  "Slivers" section to the cookbook as well documenting these patterns.

What do you think? Thoughts on where it should go?

**Comment by Sfshaza** [comment]
@brianegan, sorry, I didn't see your response until now.

I would call this an Advanced List example, since the goal is to spread out the contents and the Sliver is an implementation detail.  If we have other Sliver examples, we could create a Slivers section, then cross link to this list example. Otherwise, if folks search the site for Sliver, they should find this example. What do you think?

**Comment by brianegan** [comment]
Sounds good to me! Yep, I liked "Advanced Lists" better for the exact same reason and because I've seen non-native speakers get confused by the term Sliver. I'll go that route.

**Comment by Sfshaza** [comment]
sgtm

**Comment by pedromassangocode** [comment]
Does this still a valid issue? Don't know where to verify that.
